IMDbApp/app/src/main/java/com/danielcano/imdbapp/domainlayer/usecases/GetMoviesForUICaseImpl.kt

```diff
@@ -0,0 +1,10 @@
+package com.danielcano.imdbapp.domainlayer.usecases
+
+import com.danielcano.imdbapp.datalayer.repositories.MoviesRepository
+import com.danielcano.imdbapp.domainlayer.models.MovieModel
+
+class GetMoviesForUICaseImpl(private val repository: MoviesRepository):GetMoviesForUICase {
+    override suspend fun getMovies(): List<MovieModel> {
```

Comment:
Este comentario es solo para que tenga el conocimiento, mas no es necesario hacer un cambio. Los casos de uso, suelen implementarse con el operador invoke, que permite hacer un llamado al objeto y de forma implícita se llama la funcion.

https://kotlinlang.org/docs/operator-overloading.html#invoke-operator
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/invoke.html
https://chrynan.codes/invoking-usecases-the-kotlin-way/